csm/core/controllers/information.py

```diff
@@ -77,3 +77,115 @@ async def post(self):
             Log.error(f"Error in checking compatability: {e}")
             raise CsmInternalError(f"{e}")
         return CsmResponse(response)
+
+@CsmAuth.public
+@CsmView._app_routes.view("/api/v2/system/topology")
+class TopologyView(CsmView):
+    """
+    Complete CORTX topology will be provided
+    GET: Complete CORTX topology
+    """
+    def __init__(self, request):
+        super().__init__(request)
+        self._service = self.request.app[const.INFORMATION_SERVICE]
+
+    async def get(self):
+        """GET REST implementation for complete cortx topology."""
+        Log.info("Fetching cortx topology.")
+        # Call Cortx Information Service
+        response = await self._service.get_topology()
+        return CsmResponse(response)
+
+@CsmAuth.public
+@CsmView._app_routes.view("/api/v2/system/topology/{resource}")
+class ResourcesTopologyView(CsmView):
+    """
+    Get information about all resources from cortx topology.
+    GET: Get information about all resources from cortx topology.
+    """
+    def __init__(self, request):
+        super().__init__(request)
+        self._service = self.request.app[const.INFORMATION_SERVICE]
+
+    async def get(self):
+        """GET REST implementation to query information about all resources from cortx topology."""
+        Log.debug("Handling GET request to query information about all resources from cortx topology.")
+        resource = self.request.match_info[const.ARG_RESOURCE]
+        # Call Cortx Information Service
+        Log.info(f"Fetching cortx topology for resource:{resource}.")
+        response = await self._service.get_resources(resource)
```

Comment:
No resource/view validation error is handled here. Please add those

Reply:
Added resource/view validation error handling